salt/modules/boto_iam.py

```diff
@@ -153,6 +153,297 @@ def role_exists(name, region=None, key=None, keyid=None, profile=None):
+    try:
+        info = conn.get_group(group_name)
+        if not info:
+            return False
+        return info
+    except boto.exception.BotoServerError as e:
+        log.debug(e)
+        msg = 'Failed to get group {0} info .'
+        log.error(msg.format(group_name))
+        return False
+
+
+def add_user_to_group(user_name, group_name, region=None, key=None, keyid=None,
+                      profile=None):
+    '''
+    Add user to group
+
+    CLI example::
+
+        salt myminion boto_iam.add_user_to_group myuser mygroup
+    '''
+    group = __salt__['boto_iam.get_group'](group_name=group_name,
+                                           region=region, key=key,
+                                           keyid=keyid, profile=profile)
+    if group:
+        for _users in group['get_group_response']['get_group_result']['users']:
+            if user_name == _users['user_name']:
+                msg = 'Username : {0} is already in group {1} .'
+                log.info(msg.format(user_name, group_name))
+                return 'Exists'
+    user = __salt__['boto_iam.get_user'](user_name, region, key, keyid, profile)
+    if not group or not user:
+        msg = 'Username : {0} or group {1} do not exist .'
+        log.error(msg.format(user_name, group_name))
+        return False
+    conn = _get_conn(region, key, keyid, profile)
+    try:
+        info = conn.add_user_to_group(group_name, user_name)
+        if not info:
+            return False
+        return info
+    except boto.exception.BotoServerError as e:
+        log.debug(e)
+        msg = 'Failed to add user {0} to group {1} .'
+        log.error(msg.format(user_name, group_name))
+        return False
+
+
+def put_group_policy(group_name, policy_name, policy_json, region=None, key=None,
+                     keyid=None, profile=None):
+    '''
+    Adds or updates the specified policy document for the specified group.
+
+    CLI example::
+
+        salt myminion boto_iam.put_group_policy mygroup policyname policyrules
+    '''
+    group = __salt__['boto_iam.get_group'](group_name=group_name, region=region,
+                                           key=key, keyid=keyid, profile=profile)
+    if group:
+        conn = _get_conn(region, key, keyid, profile)
+        try:
+            created = conn.put_group_policy(group_name, policy_name,
+                                            policy_json)
+            if created:
+                log.info('Created policy for group {0}.'.format(group_name))
+                return True
+            msg = 'Could not create policy for group {0}'
+            log.error(msg.format(group_name))
+            return False
+        except boto.exception.BotoServerError as e:
+            log.debug(e)
+            msg = 'Failed to create policy for group {0}'
+            log.error(msg.format(group_name))
+            return False
+    else:
+        log.error('Group {0} does not exist'.format(group_name))
+        return False
```

Comment:
Having multiple returns of the same value is a not very nice tone. Consider this:
```python
if group:
    ...
    try:
        ...
        if created
            return True
    except ....:
        log. ....
return False